src/compiler/parser.ts

```diff
@@ -1086,10 +1086,19 @@ namespace ts {
             return currentToken;
         }
 
-        function nextToken(): SyntaxKind {
+        function nextTokenWithoutCheck() {
             return currentToken = scanner.scan();
         }
 
+        function nextToken(): SyntaxKind {
+            // if the keyword had an escape
+            if (isKeyword(currentToken) && (scanner.hasExtendedUnicodeEscape() || (scanner.getTokenValue() && scanner.getTokenText() !== scanner.getTokenValue()))) {

```

Comment:
I'd rather us track that a normal Unicode escape sequence (i.e. `\u0061`) was seen, as `scanner.getTokenText()` performs a call to `text.substring` every time it is called. This seems like unnecessary overhead to check for every keyword we encounter.

Reply:
Speaking of which, #23801 has to track similar information. See if you can sync up there.

Reply:
#23801 needs yet another token flag for marking tokens which had an incomplete escape sequence~ Similar, but no overlap.

Reply:
I'll take a look🙋🏻‍♂️